### DISCUSSION TITLE
Attempting to nuild Lora APRS iGate and Tracket. missing header files

### DISCUSSION THREAD

**Discussion by rkulzer** [lora-aprs/LoRa_APRS_iGate#323] [Q&A]
I tried following the respective Quick Start Guides for building each of the Lora APRS iGates and Tracker.

For the Tracker, I get a compile error because my platformio environment is missing the "stdint.h" header file.

For the iGate,, the "<map>" header file is missing.

Any suggestions on how to include the missing file into the platformio would be appreciated.

Thank you.

Randy Kulzer, N2CUG
rkulzer@usa.net
kyfhollc@aol.com

**Comment by peterus** [comment]
Hi Randy! Looks like you where to quick to use PlatformIO. De-Install PlatformIO first, restart VS Code, install PlatformIO again BUT now wait until the installation process was complete! A small box in the bottom right corner will print you this. Than click on restart to restart VS Code.
While the installation process is running, do NOT touch something! It will take some time ;)
Let me know if it is working now.

**Comment by rkulzer** [comment]
Peter:
Did what you said. Still having issues. Enclosed, please find output of attempt to build "LoRa_APRS_iGate-master" with env:ttgo-lora32-v1 board:
*  Executing task: C:\Users\Randy\.platformio\penv\Scripts\platformio.exe run --environment ttgo-lora32-v1 
Processing ttgo-lora32-v1 (board: esp32doit-devkit-v1; platform: espressif32 @ 6.3.1; framework: arduino)-------------------------------------------------------------------------------------------------------------------Verbose mode can be enabled via `-v, --verbose` optionCONFIGURATION: https://docs.platformio.org/page/boards/espressif32/esp32doit-devkit-v1.htmlPLATFORM: Espressif 32 (6.3.1) > DOIT ESP32 DEVKIT V1HARDWARE: ESP32 240MHz, 320KB RAM, 4MB FlashDEBUG: Current (cmsis-dap) External (cmsis-dap, esp-bridge, esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)PACKAGES: - framework-arduinoespressif32 @ 3.20009.0 (2.0.9) - tool-esptoolpy @ 1.40501.0 (4.5.1) - toolchain-xtensa-esp32 @ 8.4.0+2021r2-patch5LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldfLDF Modes: Finder ~ deep+, Compatibility ~ softFound 50 compatible librariesScanning dependencies...Dependency Graph|-- ArduinoJson @ 6.21.2       |-- AXP202X_Library @ 1.1.3    |-- APRS-Decoder-Lib @ 0.0.6   |-- esp-logger @ 1.0.0|-- ESP-FTP-Server-Lib @ 0.13.0|-- PubSubClient @ 2.8.0       |-- TinyGPSPlus @ 1.0.3        |-- OneButton @ 1.5.0|-- RadioLib @ 6.0.0|-- APRS-IS|-- WiFi @ 2.0.0|-- BoardFinder|-- Ethernet @ 2.0.0|-- SPI @ 2.0.0|-- Wire @ 2.0.0|-- System|-- Display|-- ConfigurationManagement|-- PowerManagement|-- FS @ 2.0.0|-- NTPClient|-- ArduinoOTA @ 2.0.0|-- Update @ 2.0.0|-- TimeLib|-- SPIFFS @ 2.0.0Building in release modeCompiling .pio\build\ttgo-lora32-v1\src\LoRa_APRS_iGate.cpp.oCompiling .pio\build\ttgo-lora32-v1\src\TaskAprsIs.cpp.oCompiling .pio\build\ttgo-lora32-v1\src\TaskBeacon.cpp.oCompiling .pio\build\ttgo-lora32-v1\src\TaskDisplay.cpp.oIn file included from C:/Users/Randy/.platformio/packages/framework-arduinoespressif32/cores/esp32/Arduino.h:24,                 from .pio/libdeps/ttgo-lora32-v1/esp-logger/src/logger.h:4,                 from src/TaskAprsIs.cpp:1:c:\users\randy\.platformio\packages\toolchain-xtensa-esp32\lib\gcc\xtensa-esp32-elf\8.4.0\include\stdint.h:9:16: fatal error: stdint.h: No such file or directory
***************************************************************** Looking for stdint.h dependency? Check our library registry!** CLI  > platformio lib search "header:stdint.h"* Web  > https://registry.platformio.org/search?q=header:stdint.h*****************************************************************
 # include_next <stdint.h>                ^~~~~~~~~~compilation terminated.src/LoRa_APRS_iGate.cpp:2:10: fatal error: map: No such file or directory #include <map>          ^~~~~compilation terminated.*** [.pio\build\ttgo-lora32-v1\src\TaskAprsIs.cpp.o] Error 1In file included from C:/Users/Randy/.platformio/packages/framework-arduinoespressif32/cores/esp32/Arduino.h:24,                    from .pio/libdeps/ttgo-lora32-v1/esp-logger/src/logger.h:4,                 from src/TaskBeacon.cpp:1:c:\users\randy\.platformio\packages\toolchain-xtensa-esp32\lib\gcc\xtensa-esp32-elf\8.4.0\include\stdint.h:9:16: fatal error: stdint.h: No such file or directory
***************************************************************** Looking for stdint.h dependency? Check our library registry!** CLI  > platformio lib search "header:stdint.h"* Web  > https://registry.platformio.org/search?q=header:stdint.h*****************************************************************
 # include_next <stdint.h>                ^~~~~~~~~~compilation terminated.*** [.pio\build\ttgo-lora32-v1\src\LoRa_APRS_iGate.cpp.o] Error 1*** [.pio\build\ttgo-lora32-v1\src\TaskBeacon.cpp.o] Error 1In file included from C:/Users/Randy/.platformio/packages/framework-arduinoespressif32/cores/esp32/Arduino.h:24,                 from .pio/libdeps/ttgo-lora32-v1/esp-logger/src/logger.h:4,                 from src/TaskDisplay.cpp:1:c:\users\randy\.platformio\packages\toolchain-xtensa-esp32\lib\gcc\xtensa-esp32-elf\8.4.0\include\stdint.h:9:16: fatal error: stdint.h: No such file or directory
***************************************************************** Looking for stdint.h dependency? Check our library registry!** CLI  > platformio lib search "header:stdint.h"* Web  > https://registry.platformio.org/search?q=header:stdint.h*****************************************************************
 # include_next <stdint.h>                ^~~~~~~~~~compilation terminated.*** [.pio\build\ttgo-lora32-v1\src\TaskDisplay.cpp.o] Error 1=========================================== [FAILED] Took 20.03 seconds ===========================================Environment     Status    Duration--------------  --------  ------------ttgo-lora32-v1  FAILED    00:00:20.029====================================== 1 failed, 0 succeeded in 00:00:20.029 ====================================== *  The terminal process "C:\Users\Randy\.platformio\penv\Scripts\platformio.exe 'run', '--environment', 'ttgo-lora32-v1'" terminated with exit code: 1.  *  Terminal will be reused by tasks, press any key to close it. 
Perhaps I'm not getting a copy of the github repository correctly.
I expect that solving this issue may also solve a similar issue with building the "LoRa_APRS_iGate-master".

Randy Kulzer, ***@***.***@aol.com
    On Monday, August 14, 2023 at 12:12:45 PM EDT, Peter Buchegger ***@***.***> wrote:  
 
 


Hi Randy! Looks like you where to quick to use PlatformIO. De-Install PlatformIO first, restart VS Code, install PlatformIO again BUT now wait until the installation process was complete! A small box in the bottom right corner will print you this. Than click on restart to restart VS Code.
While the installation process is running, do NOT touch something! It will take some time ;)
Let me know if it is working now.

**Comment by rkulzer** [comment]
Messer. Buchegger :
After several false starts (my fault) everything compiled and uploaded to the tracker and iGate boards.
Vielen Dank!

Randy Kulzer, ***@***.***@aol.com
    On Monday, August 14, 2023 at 12:12:45 PM EDT, Peter Buchegger ***@***.***> wrote:  
 
 


Hi Randy! Looks like you where to quick to use PlatformIO. De-Install PlatformIO first, restart VS Code, install PlatformIO again BUT now wait until the installation process was complete! A small box in the bottom right corner will print you this. Than click on restart to restart VS Code.
While the installation process is running, do NOT touch something! It will take some time ;)
Let me know if it is working now.
